src/libmusig2.c

```diff
@@ -245,22 +254,53 @@ int musig2_aggregate_pubkey(musig2_context *mc, secp256k1_pubkey *pk_list, int n
     return 1;
 }
 
-int musig2_aggregate_R(musig2_context *mc, secp256k1_pubkey batch_list[][V]) {
+int musig2_aggregate_R(musig2_context *mc, secp256k1_pubkey batch_list[][V], int state) {
 
     int i, j;
     secp256k1_pubkey* temp_R_list[mc->nr_signers];
 
     /* Aggregate the batch commitments for current message */
     for (j = 0; j < V; j++) {
+        mc->aggr_R_list[state * V + j] = malloc(sizeof (secp256k1_pubkey));
         for (i = 0; i < mc->nr_signers; i++)
             temp_R_list[i] = &batch_list[i][j];
 
-        if (!secp256k1_ec_pubkey_combine(mc->ctx, &mc->aggr_R_list[j], (const secp256k1_pubkey *const *) temp_R_list, mc->nr_signers))
+        if (!secp256k1_ec_pubkey_combine(mc->ctx, mc->aggr_R_list[state * V + j], (const secp256k1_pubkey *const *) temp_R_list, mc->nr_signers))
             return 0;
     }
     return 1;
 }
 
+int musig2_signer_precomputation(musig2_context *mc, secp256k1_pubkey *pk_list, unsigned char *serialized_batch_list, int nr_signers, int nr_messages){
```

Comment:
Given that the batch_list is sent serialised, for consistency, I would do the same for the pk_list. 
Also the pks will be sent through the wire by the different signers, to it makes sense to initialise the signer with a serialised list of pks.